services/reventService.js

```diff
@@ -0,0 +1,52 @@
+const { NEXT_PUBLIC_API_URL } = process.env;
+const GET = "GET";
+const POST = "POST";
+
+export async function fetchEndpoint(endpoint, method, ...args) {
+  if (method == GET) {
```

Comment:
the only thing that is different when we have GET or POST is the body, so we should only change the body inside the if, and if there is a body add it to the message. Then, make the request as usual